core/trino-main/src/main/java/io/trino/execution/TaskManagerConfig.java

```diff
@@ -603,4 +617,13 @@ public void applyFaultTolerantExecutionDefaults()
     {
         taskConcurrency = 8;
     }
+
+    public enum PagePartitioningStrategy
```

Comment:
I feel like user will be super confused to choose any of these. I don't know how to choose any of these either. Why do we need to expose feature config for such a detail? Does it mean we have a problem with current approach in general case?

Reply:
Generally it shouldn't be changed if the number of partitions and the size of the buffer is relatively small (e.g.: <100). However for high number of partitions (e.g.: 1000) the overhead of effectively doubling the memory usage could be quite significant. 

We were discussing this issue with @lukasz-stec offline, and there's a chance we may be able to remove the row based partitioning completely. But for that we need to be able to test columar vs row-wise with the high number of partitions. 

So for now this setting is useful for both, ensuring the memory footprint is predictable and allowing switching different strategies easily for testing.

Reply:
This property values are not self explanatory. Could it be changed automatically based on FTE number of partitions?
Do we need 3 or 4 settings for this property?

Reply:
I'm not sure it should ever be changed. This property is experimental at the moment. I may not be ever necessary to change the default. I would prefer to refrain from documenting it just yet. And since we have no empirical data yet on the impact of a strategy chosen i don't think we are ready to make any automatic assumptions.